packages/create-sitecore-jss/src/templates/nextjs/src/lib/extract-path/index.ts

```diff
@@ -0,0 +1,36 @@
+import { ParsedUrlQuery } from 'querystring';
+import * as plugins from 'temp/extract-path-plugins';
+
+export interface Plugin {
+  /**
+   * A function which will be called during path extraction
+   */
+  exec(path: string): string;
+}
+
+export class PathExtractor {
+  /**
+   * Allow plugins to extract normalized path
```

Comment:
I'd keep this more generic (there's more work done in here than just the plugins)
```suggestion
   * Extract normalized Sitecore item path
```